quranHome/src/commonMain/kotlin/com/islamversity/quran_home/QuranHomePresenter.kt

```diff
@@ -0,0 +1,22 @@
+package com.islamversity.quran_home
+
+import com.islamversity.core.mvi.BasePresenter
+import com.islamversity.core.mvi.BaseState
+import com.islamversity.core.mvi.MviProcessor
+
+class QuranHomePresenter(
```

Comment:
there is a filterFunction
I suggest filtering `Initial` Intent to avoid passing multipletimes